apps/emqx_machine/src/emqx_machine.erl

```diff
@@ -0,0 +1,181 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_machine).
+
+-export([ start/0
+        , graceful_shutdown/0
+        , is_ready/0
+        ]).
+
+-export([ stop_apps/1
+        , ensure_apps_started/0
+        ]).
+
+-export([sorted_reboot_apps/0]).
+
+-include_lib("emqx/include/logger.hrl").
+
+%% @doc EMQ X boot entrypoint.
+start() ->
+    os:set_signal(sighup, ignore),
```

Comment:
ignore sighup is an unrelated change or it is required?

Reply:
it's unrelated to graceful shutdown